Underslung Grenade Launcher, Changes to Secondary_Weapons, Vendors

5crownik007 opened the pull request:
🆑 5crownik007
rscadd: 40mm Grenade launcher is now available for use. It can be found in Attachment Vendors. 40mm comes in 5 varieties, Practice, High Explosive, Frag, Smoke and Illumination. Illumination shells are currently nonfunctional.
bugfix: Covenant Smokebomb is now in its vendor
bugfix: Changes to secondary weapon code mean they can be unloaded manually and now eject spent casings 
bugfix: Rubber M7 SMG Magazine now uses appropriate sprites. M7 bullets now do 20 shield damage instead of 10
/🆑

5crownik007 committed:
Adds smokebomb/covenant to covvendors

yeah i forgot to add the smoke bomb which i made for the covenant to the covenant weapon vendors what are you gonna do

5crownik007 committed:
Makes Covenant Smoke Purple

If it's not purple then it's not covenant is it

5crownik007 committed:
Reformat Vendor Lists, Add 40mm Grenades

Puts the lists in a nice readable vertical format instead of a disgusting, annoying to read horizontal format.

Also adds 40mm grenades and the underbarrel launcher for the MA5B/MA3 in UNSC attachment vendor and SOE attachment vendor

5crownik007 committed:
Adds 40mm Grenades

Adds 40mm grenades for use in the underbarrel launcher

5crownik007 committed:
Adds 40mm Grenade Sprites

Adds sprites for 40mm grenades for use in underbarrel launcher

5crownik007 committed:
MA5/MA3 Underbarrel Launcher

Makes attachment profile for underbarrel launcher actually work

5crownik007 committed:
Changes to Secondary Weapon Functionality

Secondary Weapons can now eject cases or store them for manual ejection. Also renamed some vars because they seemed a bit long winded/redundant.

Adds the M301 Grenade launcher

5crownik007 committed:
Allows secondary weapons to be unloaded

Checks if the user is unloading their secondary weapon instead of their primary weapon so people can actually unload their underbarrels

5crownik007 committed:
Revert "Makes Covenant Smoke Purple"

This reverts commit fc87d1dbe921ca72cadf10038f199260f8aced89.

5crownik007 committed:
40mm sounds

Gives 40mm a launch sound(should give them a unique one at some point)
Gives 40mm frag an explosion sound
Gives 40mm underbarrel a reload sound

5crownik007 committed:
Fixes reformat errors

Added a missing comma and removed an extra paren

5crownik007 committed:
Fixes 40mm sound

I misspelled sound

5crownik007 committed:
Ammo Changes

40mm arming ranges fixed
40mm frag now fragments properly
gave the SMG bullet 20 shield damage instead of 10 at XO's suggestion

5crownik007 committed:
Attach profile changes

removes erroneous list entry from underslung shotgun
underslung grenade launcher increases dispersion by 0.5(same as underslungshotgun)
gives MA3 pixel offsets for grenade launcher
gives MA3 att mods for grenade launcher

5crownik007 committed:
Secondary Weapon Sounds

Secondary weapons may now have an eject sound which is played when manually unloaded.

5crownik007 committed:
secondary sound changes

weapons can now have a sound of their own that takes priority over the sound of their ammunition

5crownik007 committed:
sound for 40mm grenade

5crownik007 committed:
40mm grenade sound

now it doesn't sound like an M41 SSR going off

5crownik007 committed:
list mistake

damned missing commas

5crownik007 committed:
comma fix

missed a comma in playsound for ejection

5crownik007 committed:
last fixes to lists, vars & sounds

it compiled successfully

5crownik007 commented:
Seems about right to me

X0-11 commented on code/modules/halo/weapons/ammo.dm:
```diff
@@ -445,3 +445,95 @@
+	armor_penetration = 5
+
+/obj/item/weapon/grenade/smokebomb/New()
+	..()
```
What's this here for?

5crownik007 replied:
Appears to be legacy code from me experimenting with smokebombs
I'll remove it real quick

5crownik007 committed:
removes erroneous smoke code

had a bit of code copied from smokebombs that was a remnant from code experiments for the 40mm smoke.

5crownik007 commented:
All checks passed, seems ready to merge

5crownik007 committed:
40mm smoke fix

40mm smoke actually produces smoke

bloxgate commented:
Changelog needs some changes, info sent on Discord

5crownik007 commented:
> 
> 
> Changelog needs some changes, info sent on Discord

Whoops I went to sleep immediately after I made my last changes

Should be good now
